src/containers/community/CommunityJoin.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import { connect } from 'react-redux'
+const { func, bool, string } = React.PropTypes
+import { get, isEmpty } from 'lodash'
+import { JOIN_COMMUNITY_WITH_CODE, resetError, joinCommunityWithCode, navigate, resetCommunityValidation, validateCommunityAttribute } from '../../actions'
+import cx from 'classnames'
+
+@connect(({communityValidation, errors}) => {
+  var error
+  if (errors[JOIN_COMMUNITY_WITH_CODE]) {
+    error = 'We couldn\'t find that code, please check it and try again'
+  }
+  let codeInvalid = !get(communityValidation, 'beta_access_code.exists')
+  return {error, codeInvalid}
+})
+export default class CommunityJoin extends React.Component {
+  static propTypes = {
+    dispatch: func,
+    codeInvalid: bool,
+    error: string
+  }
+
+  codeChanged = event => {
+    let { dispatch } = this.props
+    let { value } = event.target
+    dispatch(resetError(JOIN_COMMUNITY_WITH_CODE))
+    if (isEmpty(value.trim())) {
+      dispatch(resetCommunityValidation('beta_access_code'))
+    } else {
+      dispatch(validateCommunityAttribute('beta_access_code', value, 'exists'))
+    }
+  }
+
+  submit = () => {
+    let { dispatch, codeInvalid } = this.props
+
+    if (codeInvalid) {
+      return
+    }
+    let code = this.refs.code.value
+    dispatch(joinCommunityWithCode(code))
+    .then(action => {
+      if (action.error) {
```

Comment:
actually this can be shortened to `.then(({ error }) => error || dispatch(...))`